Logger/BaseLoggerMixins.cs

```diff
@@ -1,7 +1,44 @@
 ﻿namespace Logger
 {
-    public static class BaseLoggerMixins
+    public static class BaseLoggerMixins 
     {
-        
+        public static void Error(this BaseLogger baseLogger, string message, params string[] args)
+        { 
+            if(baseLogger is null)
+            {
+                throw new System.ArgumentNullException();
```

Comment:
For ArgumentException (and the derived ArgumentNullException) you want to call the overload that passes in the parameter name.
```suggestion
                throw new System.ArgumentNullException(nameof(baseLogger));
```